pkg/util/pki/csr.go

```diff
@@ -72,6 +47,28 @@ func IPAddressesForCertificate(crt *v1alpha2.Certificate) []net.IP {
 	return ipAddresses
 }
 
+func URIsForCertificate(crt *v1alpha2.Certificate) ([]*url.URL, error) {
+	var uris []*url.URL
+	var errs []string
+
+	for _, uriStr := range crt.Spec.URISANs {
+		uri, err := url.Parse(uriStr)
+		if err != nil {
+			errs = append(errs, err.Error())
+			continue
+		}
+
+		uris = append(uris, uri)
+	}
+
+	if len(errs) > 0 {
+		return nil, fmt.Errorf("failed to parse URIs: %s",
+			strings.Join(errs, ", "))
+	}
+
+	return uris, nil
```

Comment:
Given we don't validate that dnsNames are actual dnsNames, or common names are valid at the point where we *do* call this function, should we really be validating specifically URISANs? We *should* run these checks during resource validation (i.e. in the webhook), but I think we should do validation either on all, or none at the places where this is called 😄

Reply:
Are there any constraints on what a common name can actually be?

The reason for converting between the two for URI is that it's required for the Golang struct.

Happy to add validation for DNS names being URLs